src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2621,6 +2630,9 @@ InitializeDistTableCache(void)
 {
 	HASHCTL info;
 
+	/* make sure DistShardCacheHash was not initialized before */
+	Assert(DistShardCacheHash == NULL);
```

Comment:
I think this can happen in case of OOM.

Reply:
I don't follow how this could happen. This function is the only one making memory allocation. Once DistShardCacheHash is set, we only make few more calls that sets part of previously allocated memory.

If we are really hitting OOM due to hash_create calls within this function DistShardCacheHash should remain unset.

Could you explain more on how it could be not null ?

Reply:
An error might be thrown after DistShardCacheHash is set, but before initialisation is complete. In that case, we'll do initialisation again.

Reply:
put a try catch block outside, and did a memory reset if something bad happens